dev-haskell/hmatrix/hmatrix-0.20.2.ebuild

```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# ebuild generated by hackport 0.7.1.1
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour"
+inherit haskell-cabal
+
+DESCRIPTION="Numeric Linear Algebra"
+HOMEPAGE="https://github.com/haskell-numerics/hmatrix"
+SRC_URI="https://hackage.haskell.org/package/${P}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```

Comment:
The .cabal file still contains two flags after your patch is applied: `disable-default-paths` and `no-random_r`. If you want to have these permanently off, you will want to add a `src_configure` block that reflects this:

```
src_configure() {
	haskell-cabal_src_configure \
		--flag=-disable-default-paths \
		--flag=-no-random_r
}
```

You should also put this line underneath `# ebuild generated by hackport ...`, which will tell hackport to disable these in future bumps to the package:

```
#hackport: flags: -disable-default-paths,-no-random_r,-openblas
```

Reply:
Also, you can remove `IUSE=""` if it is empty

Reply:
You can also add these back in if you want:

```
#hackport: flags: no-random_r:no-random-r,-openblas
```

...

```
IUSE="disable-default-paths no-random-r"
```

...

```
src_configure() {
	haskell-cabal_src_configure \
		$(cabal_flag disable-default-paths disable-default-paths) \
		$(cabal_flag no-random-r no-random_r)
}
```

`metadata.xml`
```
	<use>
		<flag name="disable-default-paths">When enabled, don't add default hardcoded include/link dirs by default. Needed for hermetic builds like in nix.</flag>
		<flag name="no-random-r">When enabled, don't depend on the random_r() C function.</flag>
	</use>
```

Reply:
Something that can be helpful is generating an ebuild directly from the modified cabal file you were working on. For instance:

```bash
hackport make-ebuild dev-haskell "somepath/hmatrix-r1/hmatrix.cabal"
```

This will adjust the modified dependencies for you and make the whole process less cumbersome.

Reply:
The `hackport make-ebuild` technique was definitely the way to go.
`dos2unix` didn't appear to make any changes.